src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -321,25 +338,27 @@ private void BuildLinkTag(IDictionary<string, string> attributes, TagHelperConte
                 var attributeValue = attribute.Value;
                 if (string.Equals(attribute.Key, HrefAttributeName, StringComparison.OrdinalIgnoreCase))
                 {
-                    attributeValue = HtmlEncoder.HtmlEncode(
-                        ShouldAddFileVersion() ?
-                            _fileVersionProvider.AddFileVersionToPath(attributeValue) :
-                            attributeValue);
+                    var attributeStringValue = attributeValue as string;
+
+                    // "href" values come from bound attributes and globbing. Must always be non-null strings.
+                    Debug.Assert(attributeStringValue != null);
```

Comment:
That's fine but this is a `Debug.Assert`... We shouldn't be `Debug.Assert`ing for situations that can occur.